### PR TITLE
Claw code

### PR DESCRIPTION
**Pull request by Oshawhaat**
add code to control the claws on the robot via the left and right bumpers and triggers. Trigger to close and bumper to open

### CODE REVIEW

File: mecanumDriveFieldCentric.java [hunk 1]
```diff
@@ -12,7 +13,7 @@
 @TeleOp(name = "FieldCentricDrive")
 public class MecanumDriveFieldCentric extends LinearOpMode {
 	//SECTION - Constants
-		//SECTION - Drive Contants
+		//SECTION Drive contants
```
Comment (noonels) [inline mecanumDriveFieldCentric.java:16]: remove line

File: mecanumDriveFieldCentric.java [hunk 2]
```diff
@@ -24,6 +25,12 @@ public class MecanumDriveFieldCentric extends LinearOpMode {
 			final double ELBOW_SPEED = 0.5;
 			final double STRAFE_MULT = 1.41;
 		//!SECTION - End arm constats
+
+		//SECTION - Claw constants
+			final double CLAW_OPEN_POWER = 0.5;
```
Comment (noonels) [inline mecanumDriveFieldCentric.java:30]: looking good!

File: mecanumDriveFieldCentric.java [hunk 3]
```diff
@@ -142,13 +153,35 @@ public void runOpMode() throws InterruptedException {
...
+			if (gamepad2.right_trigger > 0) {
+				clawRight.setPower(-gamepad2.right_trigger * CLAW_CLOSE_POWER);
+				clawRightPassivePower = CLAW_CLOSE_RESIDUAL_POWER;
+			}
+			else if (gamepad2.right_bumper) {
+				clawRight.setPower(CLAW_OPEN_POWER);
+				clawRightPassivePower = 0;
+			}
+			else clawRight.setPower(-clawRightPassivePower);
+			//!SECTION End claws
```
Comment (noonels) [inline mecanumDriveFieldCentric.java:184]: Looks good!

there are simpler ways we could write these conditionals, but this is perfectly good!

**Review by scrimbles — APPROVED** [review]
ship it!

File: mecanumDriveFieldCentric.java (hunk 1 above)
Comment (qwertyuiopwm) [reply mecanumDriveFieldCentric.java:16]: Is this meant to say constants?

File: mecanumDriveFieldCentric.java [hunk 4]
```diff
@@ -41,6 +48,11 @@ public class MecanumDriveFieldCentric extends LinearOpMode {
 			private DcMotorEx MElbowRight;
 		//!SECTION - End arm motors
 
+		//SECTION - Claws
+			private CRServo clawLeft;
+			private CRServo clawRight;
+		//!SECTION - End arm motors
```
Comment (qwertyuiopwm) [inline mecanumDriveFieldCentric.java:54]: Change to "End claw setup" or alike

File: mecanumDriveFieldCentric.java (hunk 3 above)
Comment (qwertyuiopwm) [reply mecanumDriveFieldCentric.java:184]: The code could be cleaned up a tad, as a bunch of else-if statements give the code a terrible stench, but it looks good to me.

**Review by qwertyuiopwm — CHANGES_REQUESTED** [review]

File: mecanumDriveFieldCentric.java (hunk 4 above)
Comment (noonels) [reply mecanumDriveFieldCentric.java:54]: In all honesty, all of these comments for sections are extraneous, and likely just contribute to issues with readability.

The constants should be put in their own file that can be imported, so there is one place for configuration, reducing the muddiness of commits in the long run.

That having been said -- I think this is a nitpick, and not really a blocker for code like this.

Comment (qwertyuiopwm) [reply mecanumDriveFieldCentric.java:54]: Agreed

**Pull request merged**